{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "ImportError",
     "evalue": "No module named BCBio.GFF",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mImportError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-1-7631d9f2aac6>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[1;32mfrom\u001b[0m \u001b[0mBCBio\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mGFF\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mGFFExaminer\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mos\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mpandas\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mre\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mImportError\u001b[0m: No module named BCBio.GFF"
     ]
    }
   ],
   "source": [
    "from BCBio.GFF import GFFExaminer\n",
    "import os\n",
    "import pandas as pd\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "base_path = \"/Users/richard/local_data/gff3_Fall2014_intron_added/\"\n",
    "filenames = os.listdir(base_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "filename #gene #exon #intron\n",
      "Aaoar1.f 14203 39261 25058\n",
      "Aciri1_i 11202 25568 14355\n",
      "Aciri1_m 10352 23763 13408\n",
      "Acral2.f 5405 20231 14826\n",
      "Agabi_va 10438 63113 52674\n",
      "Agabi_va 11289 64626 53332\n",
      "Agahy1.f 5213 34203 28990\n",
      "Altbr1.f 10688 28593 17905\n",
      "Amamu1.f 18153 82381 64190\n",
      "Amath1.f 10354 62374 52020\n",
      "Amnli1.f 9883 26517 16634\n",
      "Amore1.f 9642 28616 18974\n",
      "Antav1.f 1488 4531 3043\n",
      "Antlo1.f 2606 2606\n",
      "problem with: intron_added_Antlo1.filtered_proteins.ExternalModels.gff3\n",
      "\n",
      "Antsi1.f 11327 68984 57657\n",
      "Apimo1.f 16992 42806 25806\n",
      "Aplpr1.f 12579 33532 20953\n",
      "Artbe1.f 7980 20652 12672\n",
      "Artel1.f 12851 73544 60693\n",
      "Artol1.f 11479 36375 24896\n",
      "Ascim1.f 17877 47727 29850\n",
      "Ascru1.f 6802 9433 2605\n",
      "Ascsa1.f 10672 32297 21625\n",
      "Ashgo1.f 4768 5045 276\n",
      "Aspac1.f 10845 35047 24194\n",
      "Aspbr1.f 5496 17620 12116\n",
      "Aspca3.f 11624 40277 28653\n",
      "Aspcam1. 5712 18519 12807\n",
      "Aspcl1.f 9121 27959 18838\n",
      "Aspfl1.f 2121 6927 4806\n",
      "Aspfo1.f 2884 9179 6292\n",
      "Aspfu1.f 9781 28608 18827\n",
      "Aspfu_A1 9916 29054 19138\n",
      "Aspgl1.f 1313 3915 2601\n",
      "Aspka1_1 11475 35684 24209\n",
      "Aspni7.f 3110 10384 7274\n",
      "Aspni_DS 14097 50390 36293\n",
      "Aspni_NR 11846 38283 26437\n",
      "Aspnid1. 10680 34743 24063\n",
      "Aspnov1. 7031 23819 16788\n",
      "Aspoch1. 8924 28780 19856\n",
      "Aspor1.f 12030 35809 23779\n",
      "Aspste1. 1721 5516 3795\n",
      "Aspsy1.f 135 413 278\n",
      "Aspte1.f 1311 4171 2860\n",
      "Asptu1.f 4857 15614 10747\n",
      "Aspve1.f 1804 5764 3959\n",
      "Aspwe1.f 2195 6947 4750\n",
      "Aspzo1.f 838 2614 1776\n",
      "Atrsp2.f 13860 75615 61755\n",
      "Aulhe2.f 12127 32217 20090\n",
      "Aurde3_1 25459 122185 96722\n",
      "Aurpu_va 7472 19075 11603\n",
      "Aurpu_va 6682 17345 10663\n",
      "Aurpu_va 11866 29461 17595\n",
      "Aurpu_va 10809 27144 16316\n",
      "Babin1.f 109 143 34\n",
      "Bacci1.f 993 4422 3429\n",
      "Batde5.f 3180 14393 11213\n",
      "Bauco1.f 10513 22548 12004\n",
      "Beaba1.f 694 1809 1115\n",
      "Bjead1_1 15473 86546 71068\n",
      "Blugr1.f 6470 17899 11429\n",
      "Boled1.f 16933 82645 65711\n",
      "Botbo1.f 16526 92254 75714\n",
      "Botci1.f 16447 43357 26910\n",
      "Botdo1.f 14998 50095 35097\n",
      "Bysci1.f 4700 12617 7917\n",
      "Cadsp1.f 22766 69285 46519\n",
      "Canar1.f 5861 7828 1928\n",
      "Canca1.f 4657 5603 918\n",
      "Canta1.f 5895 7316 1410\n",
      "Cante1.f 5533 6706 1173\n",
      "Catan1.f 14188 35442 21223\n",
      "Cenge3.f 14748 43156 28408\n",
      "Cersu1.f 12125 80856 68730\n",
      "Cerun2.f 12966 77141 64175\n",
      "Cerzm1.f 5886 13532 7646\n",
      "Chagl_1. 398 1287 889\n",
      "Chalo1.f 8505 26311 17806\n",
      "Chove1.f 12978 37930 24952\n",
      "Clafu1.f 14127 30740 16613\n",
      "Clagr3.f 11389 34527 23138\n",
      "Clapy1.f 15130 85828 70698\n",
      "Cocca1.f 6480 16727 10247\n",
      "CocheC4_ 6759 18075 11315\n",
      "CocheC5_ 13336 35437 22101\n",
      "Cocim1.f 9910 32304 22394\n",
      "Coclu2.f 12131 31591 19458\n",
      "Cocmi1.f 12007 31759 19752\n",
      "Cocsa1.f 4693 12392 7697\n",
      "Cocvi1.f 8194 21181 12987\n",
      "Coere1.f 7347 11129 3780\n",
      "Colgr1.f 12006 32967 20961\n",
      "Colhi1.f 4734 12297 7563\n",
      "Conco1.f 10635 29531 18889\n",
      "Conli1.f 12318 36736 24418\n",
      "Conlig1. 13657 36216 22559\n",
      "Conpu1.f 13761 84144 70383\n",
      "Copci1.f 13393 75796 62403\n",
      "Copci_Am 14245 77019 62774\n",
      "Corca1.f 17166 44554 27388\n",
      "Corgl3.f 20377 102875 82497\n",
      "Cormi1.f 9651 28872 19221\n",
      "Croqu1.f 6941 31337 24360\n",
      "Cryne_H9 6967 42284 35317\n",
      "Cryne_JE 6475 41896 35421\n",
      "Crypa2.f 4931 14186 9255\n",
      "Cryvi1.f 7232 45186 37954\n",
      "Cucbe1.f 12439 33752 21313\n",
      "Cybja1.f 6038 8125 2087\n",
      "Cylto1.f 13940 72058 58118\n",
      "Dacsp1.f 10242 49500 39255\n",
      "Daequ1.f 12199 70773 58574\n",
      "DalEC12_ 11173 32240 21043\n",
      "Debha1.f 6272 6739 467\n",
      "Dekbr2.f 5636 7664 2018\n",
      "Delco1.f 789 2481 1692\n",
      "Denbi1.f 33645 171165 137520\n",
      "Dicsq1.f 12290 71796 59503\n",
      "Didex1.f 12394 30433 18003\n",
      "Diocr1.f 15948 85429 69481\n",
      "Disac1.f 10299 22319 12020\n",
      "Dotse1.f 12580 28937 16357\n",
      "Dotsy1.f 11790 31892 20102\n",
      "Enccu1.f 1996 2011 15\n",
      "Enche1.f 1847 1883 36\n",
      "Encin1.f 1833 1847 14\n",
      "Encro1.f 1831 1864 33\n",
      "Entbi1.f 3632 3632\n",
      "problem with: intron_added_Entbi1.filtered_proteins.ExternalModels.gff3\n",
      "\n",
      "Eurhe1.f 8780 27012 18214\n",
      "Eutla1.f 11685 33978 22293\n",
      "Exova1.f 7453 20809 13356\n",
      "Fibra1.f 9262 60105 50843\n",
      "Fibsp1.f 32946 152694 119748\n",
      "Fishe1.f 11244 64339 53095\n",
      "Fomme1.f 11333 68698 57364\n",
      "Fompi3.f 13885 77240 63352\n",
      "Fusfu1.f 14813 41578 26765\n",
      "Fusgr1.f 4547 12674 8127\n",
      "Fusox1.f 1790 4949 3159\n",
      "Fusve1.f 12512 34851 22339\n",
      "Galma1.f 21461 113698 92176\n",
      "Ganpr1.f 13902 77525 63608\n",
      "Gansp1.f 12910 75170 62256\n",
      "Glalo1.f 13083 38836 25753\n",
      "Gloac1.f 15777 44590 28813\n",
      "Gloci1.f 2276 6231 3954\n",
      "Gloin1.f 396 1382 986\n",
      "Glotr1_1 11846 72787 60941\n",
      "Grocl1.f 8312 23727 15415\n",
      "Gymau1.f 9106 28388 19282\n",
      "Gymch1.f 18193 100221 82028\n",
      "Gymci1_1 9779 29268 19489\n",
      "Gymlu1.f 22057 116032 93952\n",
      "Gyrli1.f 11779 67772 55993\n",
      "Hanpo2.f 5177 6226 1049\n",
      "Hanva1_1 4800 5776 954\n",
      "Hebcy2.f 15382 80629 65230\n",
      "Hetan2.f 13383 72148 58763\n",
      "Hetpy1.f 7658 46555 38897\n",
      "Hisca1.f 9251 30599 21348\n",
      "Hydpi2.f 13270 77435 64148\n",
      "Hypbu1.f 6002 7331 1316\n",
      "HypCI4A_ 11712 33212 21476\n",
      "HypCO275 4461 13029 8562\n",
      "HypEC38_ 12261 35018 22748\n",
      "Hypsu1.f 17911 94703 76784\n",
      "Hyspu1.f 12352 39057 26705\n",
      "Ilysp1.f 14292 41340 27048\n",
      "Jaaar1.f 16419 90839 74409\n",
      "Karrh1.f 15043 40718 25675\n",
      "Klula1.f 5076 5251 175\n",
      "Lacam1.f 21066 94674 73587\n",
      "Lacbi2.f 23132 122070 98914\n",
      "Laesu1.f 13774 78786 65012\n",
      "Lenfl1.f 13233 35552 22319\n",
      "Lepmu1.f 12469 35201 22732\n",
      "Leppa1.f 13870 41837 27967\n",
      "Leugo1.f 5420 30295 24875\n",
      "Leumo1.f 14619 86061 71442\n",
      "Lichy1.f 2904 14978 12074\n",
      "Lipst1_1 8192 23318 15103\n",
      "Lopma1.f 7306 20438 13132\n",
      "Lopmy1.f 15153 43632 28479\n",
      "Macan1.f 12954 32261 19307\n",
      "Macfu1.f 15801 85149 69348\n",
      "Macph1.f 13806 34573 20767\n",
      "Maggr1.f 11054 30705 19651\n",
      "Malgl1.f 4286 6377 2091\n",
      "Malsy1_1 3517 6971 3454\n",
      "Melbi2.f 3040 9089 6049\n",
      "Mellp1.f 3285 15844 12559\n",
      "Melpu1.f 15881 43935 28054\n",
      "Melsp1.f 608 1724 1116\n",
      "Melva1.f 20389 61938 41549\n",
      "Metac1.f 6046 16297 10251\n",
      "Metan1.f 10583 28326 17743\n",
      "Metbi1.f 5851 7408 1531\n",
      "Micca1.f 8915 28611 19696\n",
      "Mixos1.f 6903 31373 24461\n",
      "Monha1.f 10959 36264 25305\n",
      "Monpe1_1 13560 46187 32627\n",
      "Monpu1.f 8918 28491 19573\n",
      "Monru1.f 5380 17128 11748\n",
      "Morco1.f 10909 38272 27363\n",
      "Morel1.f 2505 8821 6316\n",
      "Mucci2.f 5075 18947 13872\n",
      "Mycfi2.f 13107 47511 34402\n",
      "Mycgr3.f 10933 28552 17619\n",
      "Myrdu1.f 10685 25327 14642\n",
      "Nadfu1.f 5657 8862 3201\n",
      "Naifl1.f 13270 70814 57544\n",
      "Necha2.f 9665 29373 19708\n",
      "Nempa1.f 2661 2663 2\n",
      "Neofi1.f 10406 30984 20578\n",
      "Neole1.f 13164 75210 62046\n",
      "Neopa1.f 10366 32758 22392\n",
      "Neucr2.f 10785 30794 20009\n",
      "Neucr_tr 2711 7335 4618\n",
      "Neudi1.f 389 1001 612\n",
      "Neute_ma 1273 3584 2311\n",
      "Neute_ma 477 1287 810\n",
      "Nosce1.f 2060 2066 6\n",
      "Obbri1.f 13206 79539 66333\n",
      "Oidma1.f 510 1471 961\n",
      "Ompol1.f 2128 16142 14014\n",
      "Ophdi1.f 16027 41581 25554\n",
      "Ophpc1.f 8884 20072 11188\n",
      "Pacta1_2 5675 7541 1846\n",
      "Parbr1.f 7876 27450 19574\n",
      "Patat1.f 9794 29109 19315\n",
      "Paxin1.f 17968 89577 71601\n",
      "Paxru1.f 22065 84123 62006\n",
      "Penac1.f 13793 43599 29806\n",
      "Penbi1.f 13606 40576 26970\n",
      "PenbrAgR 7183 21801 14618\n",
      "Penca1.f 12374 38553 26178\n",
      "Pench1.f 11396 35513 24117\n",
      "PenchWis 13671 41997 28326\n",
      "Pendi1.f 9118 24305 15187\n",
      "Penex1.f 3749 11674 7925\n",
      "Penfe1.f 8208 25772 17564\n",
      "Pengl1.f 5720 17564 11844\n",
      "Penja1.f 6663 20819 14156\n",
      "Penla1.f 7416 23017 15601\n",
      "Penox1.f 9979 29452 19473\n",
      "Penra1.f 10210 31562 21352\n",
      "Phaal1.f 8834 24561 15727\n",
      "Phaca1.f 13937 74759 60819\n",
      "Phchr2.f 13602 73603 60001\n",
      "Phlbr1.f 16170 91546 75375\n",
      "Phlgi1.f 11891 71317 59426\n",
      "Photr1.f 8501 22996 14495\n",
      "Phybl2.f 1551 6986 5435\n",
      "Picme2.f 5546 6816 1227\n",
      "Picpa1.f 5040 5612 572\n",
      "Picst3.f 669 924 255\n",
      "Pieho1_1 7572 13903 6331\n",
      "Pilcr1.f 21583 102528 80921\n",
      "PirE2_1. 14648 45229 30577\n",
      "Pirin1.f 11767 60685 48916\n",
      "Pismi1.f 21064 85126 64041\n",
      "Pisti1.f 22701 93250 70519\n",
      "PleosPC1 12330 75035 62704\n",
      "PleosPC9 12206 74225 62015\n",
      "Plesi1.f 9079 25744 16665\n",
      "Plicr1.f 13626 79621 65978\n",
      "Pneji1.f 3520 20191 16671\n",
      "Podan2.f 1520 3669 2147\n",
      "Polar1.f 17525 92365 74840\n",
      "Polci1.f 10582 22439 11857\n",
      "Pospl1.f 9113 55872 46757\n",
      "PosplRSB 12541 78157 65610\n",
      "Psean1_1 6640 11452 4812\n",
      "Psehu1.f 7472 10953 3481\n",
      "Pucgr1.f 684 3301 2617\n",
      "Pucst1.f\n",
      "problem with: intron_added_Pucst1.filtered_proteins.ExternalModels.gff3\n",
      "\n",
      "Punst1.f 11538 71843 60304\n",
      "Pycci1.f 10442 61298 50856\n",
      "Pycco166 12798 74601 61803\n",
      "Pyrco1.f 13367 35901 22534\n",
      "Pyrtr1.f 12141 32671 20530\n",
      "Pyrtt1.f 11799 29546 17747\n",
      "Ramac1.f 19287 106579 87291\n",
      "Rhich1.f 17676 80150 62474\n",
      "Rhimi1_1 10905 43961 33056\n",
      "Rhior3.f 17467 57981 40514\n",
      "Rhiso1.f 15157 88436 73279\n",
      "Rhivi1.f 14469 80703 66233\n",
      "Rhoba1_1 7283 45433 38150\n",
      "Rhomi1.f 7814 43865 36051\n",
      "Rhyru1.f 12117 38327 26210\n",
      "Ricme1.f 18952 94316 75364\n",
      "Rozal1_1 6350 26973 20623\n",
      "Sacce1.f 6575 6860 285\n",
      "SacceM37 5974 6492 500\n",
      "SacceM38 5984 6500 494\n",
      "SacceM38 5969 6498 508\n",
      "SacceM38 5991 6519 511\n",
      "SacceM38 5989 6527 520\n",
      "SacceYB2 5994 7176 1182\n",
      "Sacpr1.f 8081 24884 16803\n",
      "Saico1.f 7034 15698 8663\n",
      "Schco3.f 16319 86053 69734\n",
      "Schco_Lo 13827 76748 62921\n",
      "Schco_Ta 15199 80095 64896\n",
      "Schcy1.f 5180 11129 5949\n",
      "Schja1.f 4878 10266 5388\n",
      "Schoc1.f 4986 10519 5533\n",
      "Schpa1.f 17098 98823 81725\n",
      "Schpo1.f 5134 10439 5305\n",
      "Sclci1.f 21012 90943 69912\n",
      "Sclsc1.f 14503 40568 26065\n",
      "Sebve1.f 15312 75701 60384\n",
      "Sepmu1.f 10233 24925 14689\n",
      "Seppo1.f 2369 6117 3720\n",
      "Serla_va 13805 75115 61308\n",
      "SerlaS7_ 14495 76165 61667\n",
      "SerlaS7_ 12789 73301 60512\n",
      "Settu1.f 11702 31062 19360\n",
      "Sisbr1.f 15653 87180 71527\n",
      "Sisni1.f 13080 77886 64806\n",
      "Sissu1.f 13657 77892 64235\n",
      "Spapa3.f 5983 7207 1223\n",
      "Sphst1.f 35274 164620 129318\n",
      "Spofi1.f 10783 29109 18326\n",
      "Spoli1.f 9062 56698 47636\n",
      "Spore1.f 6648 9710 3062\n",
      "Sporo1.f 5536 39911 34375\n",
      "Spoth2.f 1988 5649 3660\n",
      "Stano2.f 12380 32994 20614\n",
      "Stehi1.f 14072 91696 77623\n",
      "Suibr1.f 22453 101999 79546\n",
      "Suilu1.f 18316 87779 69415\n",
      "Symko1.f 10482 34636 24154\n",
      "Talma1_2 10638 33631 22993\n",
      "Talst1_2 13252 43638 30386\n",
      "Tapde1_1 4609 7568 2959\n",
      "Terbo1.f 9288 34099 24811\n",
      "Theau1.f 8798 29332 20534\n",
      "Thian1.f 358 1033 675\n",
      "Thiap1.f 338 928 590\n",
      "Thiar1.f 10954 30720 19766\n",
      "Thihy1.f 2759 7792 5033\n",
      "Thite2.f 5308 15178 9870\n",
      "Tilan2.f 6810 22632 15822\n",
      "Tralj1.f 13180 75368 62188\n",
      "Trave1.f 14296 83127 68829\n",
      "Treme1.f 8313 47812 39499\n",
      "Trepe1.f 5846 16169 10323\n",
      "Trias1.f 3979 10552 6573\n",
      "Triat2.f 11828 33788 21959\n",
      "Triha1.f 4152 11143 6990\n",
      "Trilo3.f 4925 13392 8467\n",
      "Trima3.f 22885 92270 69384\n",
      "Triol1.f 831 3073 2242\n",
      "Trire2.f 9143 27945 18802\n",
      "TrireRUT 6043 17506 11463\n",
      "Triru1.f 8707 26976 18269\n",
      "Trisp1.f 1612 9414 7802\n",
      "Triver1. 8024 20636 12612\n",
      "TriviGv2 12423 36386 23963\n",
      "Tryvi1.f 1161 3322 2161\n",
      "Tubme1.f 7496 29021 21525\n",
      "Tulca1.f 19659 91344 71680\n",
      "Umbra1.f 9931 47141 37210\n",
      "Uncre1.f 5850 18233 12383\n",
      "Ustma1.f 6522 11422 4900\n",
      "Veral1.f 2772 7845 5073\n",
      "Verda1.f 10535 29685 19150\n",
      "Veren1.f 13748 39009 25261\n",
      "Volvo1.f 3511 25602 22091\n",
      "Walic1.f 4863 16626 11763\n",
      "Walse1.f 5284 21312 16028\n",
      "Wesor1.f 10410 28893 18483\n",
      "Wican1.f 313 421 107\n",
      "Wilmi1.f 13093 42373 29278\n",
      "Wolco1.f 12746 80404 67651\n",
      "Xanpa1.f 1219 3691 2472\n",
      "Xylhe1.f 3257 10904 7647\n",
      "Yarli1.f 6447 7520 1072\n",
      "Zasce1.f 1179 3049 1870\n",
      "Zoprh1.f 7001 19861 12860\n"
     ]
    }
   ],
   "source": [
    "num_gene = []\n",
    "num_exon = []\n",
    "num_intron = []\n",
    "print \"filename\", \"#gene\", \"#exon\", \"#intron\"\n",
    "for filename in filenames:\n",
    "    examiner = GFFExaminer()\n",
    "    in_handle = open(base_path+filename)\n",
    "    test = examiner.available_limits(in_handle)\n",
    "    \n",
    "    \n",
    "    try:\n",
    "        print filename[13:21], test['gff_type'][('gene',)], test['gff_type'][('exon',)], test['gff_type'][('intron',)]\n",
    "    except:\n",
    "        print\n",
    "        print \"problem with: \"+filename\n",
    "        print\n",
    "        continue\n",
    "    num_gene.append(test['gff_type'][('gene',)])\n",
    "    num_exon.append(test['gff_type'][('exon',)])\n",
    "    num_intron.append(test['gff_type'][('intron',)])\n",
    "    in_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "statistics = pd.DataFrame({\"gene_count\":num_gene, \"exon_count\":num_exon, \"intron_count\":num_intron}, \n",
    "             columns=[\"gene_count\", \"exon_count\", \"intron_count\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "statistics.to_csv(\"exon_intron_count.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## make concise filename\n",
    "filenames_short = []\n",
    "for filename in filenames:\n",
    "    match_obj = re.match(\"intron_added_(.*?).filtered_proteins\",filename)\n",
    "    filenames_short.append(match_obj.group(1))\n",
    "\n",
    "with open(\"filenames_short.csv\",\"w\") as f:\n",
    "    for name in filenames_short:\n",
    "        f.write(name+\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "full_table_names = pd.read_csv(\"name_exon_intron_count.csv\")['name'] # name needed to add into the table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# match rna seq count to exon, intron density\n",
    "long_short_name_pair = pd.read_csv(\"../July29/long_short_name_pair_2015.csv\")\n",
    "short_long_dict = {long_short_name_pair['shortName'][i]:long_short_name_pair['longName'][i] \n",
    "                   for i in range(long_short_name_pair.shape[0])}\n",
    "\n",
    "long_rna_pair =  pd.read_csv(\"../July29/rna_seq_count.csv\")\n",
    "long_count_dict = {long_rna_pair['name'][i]:long_rna_pair[' rna_seq_count'][i] \n",
    "                   for i in range(long_rna_pair.shape[0])}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "problem with Botdo1\n",
      "problem with Hyspu1\n",
      "problem with Leugo1\n",
      "problem with Rhyru1\n",
      "problem with Terbo1\n",
      "problem with Xanpa1\n"
     ]
    }
   ],
   "source": [
    "longname = []\n",
    "shortname = []\n",
    "counts = []\n",
    "for name in full_table_names:\n",
    "    try:\n",
    "        long_name = short_long_dict[name]\n",
    "        try:\n",
    "            count = long_count_dict[long_name]\n",
    "        except:\n",
    "            count = 0\n",
    "        longname.append(long_name)\n",
    "        shortname.append(name)\n",
    "        counts.append(count)\n",
    "            \n",
    "    except:\n",
    "        print \"problem with\", name\n",
    "        longname.append(\"N/A\")\n",
    "        shortname.append(name)\n",
    "        counts.append(0)\n",
    "\n",
    "rna_seq_intron_density = pd.DataFrame({'longName':longname,'shortName':shortname,'count':counts})\n",
    "rna_seq_intron_density.to_csv(\"intron_density_rna_seq_count.csv\", index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "## filter table to exclude rna count zero species\n",
    "data = pd.read_csv(\"./longNme_shortName_exon_intron_rna_seq_count.csv\")\n",
    "idx = data['rna_seq_count'] > 0\n",
    "filtered_data = data[idx]\n",
    "filtered_data.to_csv(\"longNme_shortName_exon_intron_rna_seq_count_filtered.csv\",index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## final step, add gc\n",
    "complete_data = pd.read_csv(\"../common_file/CompleteV6.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "short_name_gc_dict = {complete_data['ShortName'][i]:complete_data['Genome_GC'][i] \n",
    "                      for i in range(complete_data.shape[0])}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "problem with Aciri1_meta\n",
      "problem with Agabi_varbisH97_2\n",
      "problem with Agabi_varbur_1\n",
      "problem with Altbr1\n",
      "problem with Aspac1\n",
      "problem with Aspca3\n",
      "problem with Aspcam1\n",
      "problem with Aspfu1\n",
      "problem with Aspni_DSM_1\n",
      "problem with Aspni_NRRL3_1\n",
      "problem with Aspnov1\n",
      "problem with Aspoch1\n",
      "problem with Aspste1\n",
      "problem with Batde5\n",
      "problem with Botci1\n",
      "problem with Cante1\n",
      "problem with Chagl_1\n",
      "problem with CocheC4_1\n",
      "problem with CocheC5_3\n",
      "problem with Conli1\n",
      "problem with Copci1\n",
      "problem with Cryne_H99_1\n",
      "problem with Fusgr1\n",
      "problem with Fusox1\n",
      "problem with HypCI4A_1\n",
      "problem with HypCO275_1\n",
      "problem with Maggr1\n",
      "problem with Mellp1\n",
      "problem with Mucci2\n",
      "problem with Mycgr3\n",
      "problem with Neucr2\n",
      "problem with Neudi1\n",
      "problem with Neute_mat_a1\n",
      "problem with Patat1\n",
      "problem with PenbrAgRF18_1\n",
      "problem with Pench1\n",
      "problem with Phybl2\n",
      "problem with Picst3\n",
      "problem with PleosPC9_1\n",
      "problem with Pospl1\n",
      "problem with Pucgr1\n",
      "problem with Pyrco1\n",
      "problem with Pyrtr1\n",
      "problem with Rhior3\n",
      "problem with SacceM3707_1\n",
      "problem with SacceM3836_1\n",
      "problem with SacceM3837_1\n",
      "problem with SacceM3838_1\n",
      "problem with SacceM3839_1\n",
      "problem with SacceYB210_1\n",
      "problem with Sclsc1\n",
      "problem with SerlaS7_3_2\n",
      "problem with SerlaS7_9_2\n",
      "problem with Spapa3\n",
      "problem with Spofi1\n",
      "problem with Treme1\n",
      "problem with Triat2\n",
      "problem with Trire2\n",
      "problem with TriviGv29_8_2\n"
     ]
    }
   ],
   "source": [
    "gc_list = []\n",
    "\n",
    "for name in filtered_data['shortName']:\n",
    "    try:\n",
    "        gc = short_name_gc_dict[name]\n",
    "        gc_list.append(gc)        \n",
    "    except:\n",
    "        print \"problem with\", name\n",
    "        gc_list.append(-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "with open(\"gc_list.csv\",'w') as f:\n",
    "    for gc in gc_list:\n",
    "        f.write(str(gc)+\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "## final filtered table\n",
    "filtered_table = pd.read_csv(\"longNme_shortName_exon_intron_rna_seq_count_filtered.csv\")\n",
    "idx = filtered_table['genome_gc'] > 0\n",
    "final_table = filtered_table[idx]\n",
    "final_table.to_csv(\"longNme_shortName_exon_intron_rna_seq_count_final.csv\",index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
