{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "'''\n",
    "1. set up temp repo for rna seq data: /storage2/meta/rnaSeqData\n",
    "\n",
    "2. write code to make rna-seq script automatically given the name of the downlaoded rna-seq file\n",
    "\n",
    "3. make table in the format shortName, RNASeqFileName, downloadURL\n",
    "\n",
    "4. the intronDensity VS gc plot is divided in to 3x3 blocks\n",
    "\n",
    "Aug 4 updated, new full info table and rna-seq info table in 1_DataSmall\n",
    "\n",
    "'''\n",
    "import pandas as pd\n",
    "fullTable = pd.read_csv(\"/home/richard/research/1_DataSmall/fullTableInfoGff3GffRNA.csv\")\n",
    "fullTable.index = fullTable['shortName']\n",
    "RNASeqFileInfo = pd.read_csv(\"/home/richard/research/1_DataSmall/rnaSeqFileList.csv\")\n",
    "dataForRNASeq = '/storage3/w/richard/meta2015/dataForRNASeq/'\n",
    "placeForRNASeq = '/storage3/w/richard/meta2015/placeForRNASeq/'\n",
    "dataPath = '/storage3/w/richard/meta2015/fasta_gff_gtf_07302015/'\n",
    "tempPath = '/storage3/w/richard/tempFile/'\n",
    "logPath = '/storage3/w/richard/meta2015/logFile/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# 1 ###################################\n",
    "## (1,1) low gc, high intron density\n",
    "# Pneji1, Pneumocystis jirovecii\n",
    "\n",
    "shortName = 'Pneji1'\n",
    "\n",
    "# # make script to download rna seq\n",
    "# script_rna = 'cd {}\\n'.format(dataForRNASeq)\n",
    "# for URL in rnaSeqFileURLs:\n",
    "#     script_rna+=\"wget {}\\n\".format(URL)\n",
    "# # print script_rna\n",
    "# with open(\"{}getRNASeqData_{}.sh\".format(logPath, shortName), 'w') as f: f.write(script_rna)\n",
    "\n",
    "\n",
    "fastaName = fullTable.loc[shortName]['fastaName']\n",
    "\n",
    "idx = (RNASeqFileInfo['shortName'] == shortName); \n",
    "rnaSeqFileNames = RNASeqFileInfo['fileName'].loc[idx]\n",
    "rnaSeqFileURLs = RNASeqFileInfo['URL'].loc[idx]\n",
    "\n",
    "# make script for the whole process of rna seq, prep -> ran-seq\n",
    "# create directory\n",
    "script = 'cd {}\\n'.format(placeForRNASeq)\n",
    "script += 'mkdir {0}\\ncd {0}\\n'.format(shortName)\n",
    "\n",
    "# prep folders\n",
    "script += 'mkdir arx data data/starg star\\n'\n",
    "\n",
    "# copy genome file and unzip it\n",
    "script += 'cp {}{}/{} ./data\\n'.format(dataPath, shortName, fastaName)\n",
    "script += 'gunzip ./data/{}\\n'.format(fastaName)\n",
    "\n",
    "# get RNA-seq \n",
    "script += 'cd arx\\n'\n",
    "for fileName in rnaSeqFileNames:\n",
    "    script += 'cp {}{} .\\n'.format(dataForRNASeq,fileName)\n",
    "    script += '/home/tool/sratoolkit/bin/fastq-dump --split-3 {}\\n'.format(fileName)\n",
    "\n",
    "# prepare genome for star\n",
    "script += 'cd ../data/starg\\n'\n",
    "script += '/home/tool/STAR/source/STAR --runMode genomeGenerate --genomeDir . --genomeFastaFiles ../{} --runThreadN 8\\n'.format(fastaName[:-3])\n",
    "\n",
    "## run rna-seq\n",
    "# generate run_star.sh\n",
    "script += \"cd ../../star\\n\"\n",
    "\n",
    "################################################################################\n",
    "################################################################################\n",
    "#######################  generate seperate file ################################\n",
    "runStar = \"nohup /home/tool/STAR/source/STAR --genomeDir ../data/starg --runThreadN 8 --readFilesIn \"\n",
    "group1, group2 = [], []\n",
    "for fileName in rnaSeqFileNames:\n",
    "    group1.append('../arx/'+fileName[:-4]+'_1.fastq')\n",
    "    group2.append('../arx/'+fileName[:-4]+'_2.fastq')\n",
    "runStar += ','.join(group1) + ' '\n",
    "runStar += ','.join(group2) + ' > loginfo &\\n'\n",
    "with open(tempPath+'run_star_'+shortName+'.sh', 'w') as f: f.write(runStar)\n",
    "################################################################################\n",
    "################################################################################\n",
    "\n",
    "# copy and execute run_star.sh\n",
    "script += 'cp {}run_star_{}.sh .\\n'.format(tempPath,shortName)\n",
    "script += 'bash run_star_{}.sh &\\n'.format(shortName)\n",
    "\n",
    "with open(\"{}rnaSeq_{}.sh\".format(logPath,shortName), 'w') as f: f.write(script)\n",
    "# print script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# ####################################\n",
    "# ## (1,3) low gc = 31.9, low intronDensity = 1.61\n",
    "# # Mycgr3, Mycosphaerella graminicola\n",
    "# shortName = 'Mycgr3'\n",
    "# # make script to download rna seq\n",
    "# script_rna = 'cd {}\\n'.format(dataForRNASeq)\n",
    "# idx = RNASeqFileInfo['shortName'] == shortName;\n",
    "# for URL in RNASeqFileInfo.loc[idx]['URL']:\n",
    "#     script_rna+=\"wget {}\\n\".format(URL)\n",
    "# print script_rna\n",
    "# # with open(\"{}getRNASeqData_{}.sh\".format(logPath, shortName), 'w') as f: f.write(script_rna)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# 2 ####################################\n",
    "# ## (1,3) low gc = 26.4, low intronDensity = 1.61\n",
    "# # Spofi1, Sporormia fimetaria\n",
    "shortName = 'Spofi1'\n",
    "\n",
    "\n",
    "fastaName = fullTable.loc[shortName]['fastaName']\n",
    "\n",
    "idx = (RNASeqFileInfo['shortName'] == shortName); \n",
    "rnaSeqFileNames = RNASeqFileInfo['fileName'].loc[idx]\n",
    "rnaSeqFileURLs = RNASeqFileInfo['URL'].loc[idx]\n",
    "\n",
    "# make script for the whole process of rna seq, prep -> ran-seq\n",
    "# create directory\n",
    "script = 'cd {}\\n'.format(placeForRNASeq)\n",
    "script += 'mkdir {0}\\ncd {0}\\n'.format(shortName)\n",
    "\n",
    "# prep folders\n",
    "script += 'mkdir arx data data/starg star\\n'\n",
    "\n",
    "# copy genome file and unzip it\n",
    "script += 'cp {}{}/{} ./data\\n'.format(dataPath, shortName, fastaName)\n",
    "script += 'gunzip ./data/{}\\n'.format(fastaName)\n",
    "\n",
    "# get RNA-seq \n",
    "script += 'cd arx\\n'\n",
    "for fileName in rnaSeqFileNames:\n",
    "    script += 'cp {}{} .\\n'.format(dataForRNASeq,fileName)\n",
    "    script += '/home/tool/sratoolkit/bin/fastq-dump --split-3 {}\\n'.format(fileName)\n",
    "\n",
    "# prepare genome for star\n",
    "script += 'cd ../data/starg\\n'\n",
    "script += '/home/tool/STAR/source/STAR --runMode genomeGenerate --genomeDir . --genomeFastaFiles ../{} --runThreadN 8\\n'.format(fastaName[:-3])\n",
    "\n",
    "## run rna-seq\n",
    "# generate run_star.sh\n",
    "script += \"cd ../../star\\n\"\n",
    "\n",
    "################################################################################\n",
    "################################################################################\n",
    "#######################  generate seperate file ################################\n",
    "runStar = \"nohup /home/tool/STAR/source/STAR --genomeDir ../data/starg --runThreadN 8 --readFilesIn \"\n",
    "group1, group2 = [], []\n",
    "for fileName in rnaSeqFileNames:\n",
    "    group1.append('../arx/'+fileName[:-4]+'_1.fastq')\n",
    "    group2.append('../arx/'+fileName[:-4]+'_2.fastq')\n",
    "runStar += ','.join(group1) + ' '\n",
    "runStar += ','.join(group2) + ' > loginfo &\\n'\n",
    "with open(tempPath+'run_star_'+shortName+'.sh', 'w') as f: f.write(runStar)\n",
    "################################################################################\n",
    "################################################################################\n",
    "\n",
    "# copy and execute run_star.sh\n",
    "script += 'cp {}run_star_{}.sh .\\n'.format(tempPath,shortName)\n",
    "script += 'bash run_star_{}.sh &\\n'.format(shortName)\n",
    "\n",
    "with open(\"{}rnaSeq_{}.sh\".format(logPath,shortName), 'w') as f: f.write(script)\n",
    "# print script"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
